addons/web/static/lib/owl/owl.js

```diff
@@ -4265,8 +4265,8 @@
             if (this.dev) {
                 // Throw error on duplicate keys in dev mode
                 this.helpers.add("OwlError");
-                this.addLine(`if (keys${block.id}.has(String(key${this.target.loopLevel}))) { throw new OwlError(\`Got duplicate key in t-foreach: \${key${this.target.loopLevel}}\`)}`);
-                this.addLine(`keys${block.id}.add(String(key${this.target.loopLevel}));`);
+                this.addLine(`if (keys${block.id}.has(JSON.stringify(key${this.target.loopLevel}))) {throw new OwlError(\`Got duplicate key in t-foreach: \${key${this.target.loopLevel}}\`)}`);

```

Comment:
To avoid this error:
![owl_error](https://user-images.githubusercontent.com/114914971/209634087-fbe5d503-3f60-45ee-a44e-21fe7e76f283.png)